### ISSUE TITLE
Cannot regenerate regression result

### ISSUE THREAD

**Issue by tbody-cfs** [cfs-energy/cfspopcon#10]
Running `poetry run python tests/regression_results/generate_regression_results.py` and then `poetry run pytest tests/test_regression_against_cases.py::test_regression_against_case\[SPARC_PRD\]` doesn't pass.

```
>       assert_allclose(dataset, reference_dataset, rtol=1e-8, atol=0)
E       AssertionError: Left and right Dataset objects are not close
E       
E       
E       Differing data variables:
E       L   radiated_power_method                 object RadiationMethod.Radas
E       R   radiated_power_method                 <U5 'Radas'
E       L   core_radiator                         object Impurity.Xenon
E       R   core_radiator                         <U5 'Xenon'
E       L   profile_form                          object ProfileForm.prf
E       R   profile_form                          <U3 'prf'
E       L   SOL_momentum_loss_function            object MomentumLossFunction.KotovRe...
E       R   SOL_momentum_loss_function            <U11 'KotovReiter'
E       L   lambda_q_scaling                      object LambdaQScaling.EichRegression15
E       R   lambda_q_scaling                      <U16 'EichRegression15'
E       L   energy_confinement_scaling            object ConfinementScaling.ITER98y2
E       R   energy_confinement_scaling            <U8 'ITER98y2'
E       L   fusion_reaction                       object ReactionType.DT
E       R   fusion_reaction                       <U2 'DT'

tests/test_regression_against_cases.py:26: AssertionError
```

**Comment by tbody-cfs** [comment]
Need to change https://github.com/cfs-energy/cfspopcon/blob/main/cfspopcon/file_io.py#L50C26-L50C26
to `if val.dtype == object or val.dtype.type == np.str_:`

**Comment by tbody-cfs** [comment]
Also fixed by wiping out `.venv` and reinstalling

**Comment by hassec** [comment]
For future reference the problem here was a result of installing `netcdf4`. 
`xarray` automatically switches from `scipy` to `netcdf4` as i/o backend if it finds `netcdf4` and this results in having strings inside the dataset encoded as `np._str` instead of `object`. 

The conversion logic to promote a string to an enum instance didn't account for this possibility, which is fixed by @tbody-cfs's suggestion above and implemented in #14

**Comment by hassec** [comment]
closed by #14
